### ISSUE TITLE
fit_summary Error KeyError: 'trial_info'

### ISSUE THREAD

**Issue by amybachir** [awslabs/autogluon#675]
I'm trying autogluon for the first time with a dataset of 75,000 data points and labels.
autogluon version 0.0.13
OS: 
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic

Here is my predictor, it is a TabularPrediction with hyperparameter_tune turned on:
`predictor = task.fit(train_data=train_data, label=label_column, output_directory=savedir, problem_type='binary', eval_metric='accuracy', hyperparameter_tune=True )`

I tried with hyperparameter_tune  off and the task finished with no problem. Once I turned on hyperparameter_tune, it errored with the following error:
```
*** Details of Hyperparameter optimization ***
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-1b709fc952c0> in <module>
     11 predictor = task.fit(train_data=train_data, label=label_column, output_directory=savedir, problem_type='binary', eval_metric='accuracy', hyperparameter_tune=True )
---> 13 results = predictor.fit_summary()
     14 
     15 # Inference time:

/usr/local/lib/python3.6/dist-packages/autogluon/task/tabular_prediction/predictor.py in fit_summary(self, verbosity)
    414                     hpo_model = hpo_results[model_type]
    415                     print("HPO for %s model:  Num. configurations tried = %s, Time spent = %s, Search strategy = %s"
--> 416                           % (model_type, len(hpo_model['trial_info']), hpo_model['total_time'], hpo_model['search_strategy']))
    417                     print("Best hyperparameter-configuration (validation-performance: %s = %s):"
    418                           % (self.eval_metric, hpo_model['validation_performance']))

KeyError: 'trial_info'

```

**Comment by jwmueller** [comment]
Thanks for pointing out this bug. I believe it should be fixed in this PR:  https://github.com/awslabs/autogluon/pull/678 
if you'd like to install that version of autogluon locally and try it out. 

Also just FYI:  `hyperparameter_tune` is currently not typically the best way to maximize accuracy in TabularPrediction. Instead consider specifying: `task.fit(..., presets='best_quality')` which utilizes stack-ensembling instead of HPO.

https://autogluon.mxnet.io/tutorials/tabular_prediction/tabular-quickstart.html#maximizing-predictive-performance

**Comment by amybachir** [comment]
> Thanks for pointing out this bug. I believe it should be fixed in this PR: #678
> if you'd like to install that version of autogluon locally and try it out.
> 
> Also just FYI: `hyperparameter_tune` is currently not typically the best way to maximize accuracy in TabularPrediction. Instead consider specifying: `task.fit(..., presets='best_quality')` which utilizes stack-ensembling instead of HPO.
> 
> https://autogluon.mxnet.io/tutorials/tabular_prediction/tabular-quickstart.html#maximizing-predictive-performance

Thanks for the tip @jwmueller ! I will try that out!
